platform/src/ToolsManager.js

```diff
@@ -70,9 +132,25 @@ class ToolManager {
             }
 
             if (xhr.status === 200) {    
-                // Rewrite URLs in tool config
-                let baseURL = toolUrl.url.substring(0, toolUrl.url.lastIndexOf("/")); // remove the name of the json file (including the trailing slash)
-                let toolConfig = xhr.responseText.replaceAll("{{BASE-URL}}", baseURL);
+
+                var toolConfig;
+                let response_text =  xhr.responseText;
+                let response_text_regexp = response_text.match(new RegExp(/{{BASE-URL}}:[0-9]*/));
+                if (response_text_regexp != null){
+                    // There is a reference to another endpoint, so base URL must be set accordingly (There is a {{BASE-URL}} placeholder plus a port).
+                    var base_url = utility.getBaseURL();
+                    let url_port = response_text_regexp[0];
+                    let path = this.fetchPathByPort(this.getPort(url_port));
+                    
+                    toolConfig = response_text.replaceAll(url_port, base_url + path);
```

Comment:
But what if there are multiple references to different endpoints, including, perhaps to this tool? I think you need to wrap your if statement into a loop and execute it separately for each match of the regexp.

Reply:
Not sure I understood this correctly but this code block is already in a loop and on each iteration we only have one url variable. If you are referring to cases where we have {{ID-somthing}} as a placeholder, these are kept intact until the program reaches the `interpolate() `method in the `ActivityManager` class.

Reply:
Code block was refactored in a method and unit tests were added too.